### PR TITLE
use local scale to make autofit in auto gamecapture mode

### PR DESCRIPTION
**Pull request by summeroff**
also need changes in osn and slobs

### CODE REVIEW

**Comment by EddyGharbi** [comment]
With these changes, am I correct saying we will auto fit to screen every time we will render frames?

**Comment by summeroff** [comment]
Even better. It not only fit to screen by default but if user change size it will rescalse games to that size.

**Comment by EddyGharbi** [comment]
Ok yeah that's where I was getting at, I think it's good that it fits the screen by default but we shouldn't force them to only fit the screen in my opinion.

**Comment by EddyGharbi** [comment]
I suggest we make the change so we don't force fitting the screen at all time. By default it should but we should let the user do what they want passed that.

**Comment by summeroff** [comment]
>>we should let the user do what they want passed that.
yes it was working like that and will be. 

By default it created with size of a screen. and autoscale any game to keep it fitting a screen. 
And when user change size to something he needs it will scale game to that new size. So capturing games what have different window sizes not break user scene alignment. 

New implementation works better. As it does not change a scale property of a scene item as in previous implementation.

File: plugins/win-capture/game-capture.c [hunk 1]
```diff
@@ -2022,6 +2006,15 @@ static void game_capture_render(void *data, gs_effect_t *effect)
...
 					     : OBS_EFFECT_OPAQUE);
+	
+	if (gc->config.mode == CAPTURE_MODE_AUTO) {	
+		struct obs_video_info ovi;	
+		obs_get_video_info(&ovi);
+		float cx_scale = ovi.base_width/(float)gc->cx;
+		float cy_scale = ovi.base_height/(float)gc->cy;
+		gs_matrix_push();
+		gs_matrix_scale3f(cx_scale, cy_scale, 1.0f);
+	}
```
Comment (EddyGharbi) [inline plugins/win-capture/game-capture.c:2049]: 2 questions:
1. Is it the action that fits the source to the screen?
2. Is it executed on each `game_capture_render`?

My concern is that these 2 are true, that would mean we are fitting to the screen all the time and we don't let the user rescale the source.

Comment (summeroff) [reply plugins/win-capture/game-capture.c:2049]: gs_matrix_scale3f is add scale transformation not overwrite it. 
So it keeps user scale and add additional one to compensate changes in resolution between different games.

Comment (EddyGharbi) [reply plugins/win-capture/game-capture.c:2049]: Got it, makes sense

**Review by EddyGharbi — APPROVED** [review]

**Pull request merged**